### PR TITLE
DM-15157: Image plotting refactor

### PR DESCRIPTION
**Pull request by robyww**
- created a framework for processing whole FITS file (fitseval)
 - improved image reading, no more 2D array
 - improved mask performance
 - broke up FITSRead into multiple files
 - created a new package (plotdata) for FitsRead related processing
 - improved cacheing but still needs more work
 - moved header processing the the client
 - recognize multiple WCS from file

_to test:_
There really are no new features. The following should work:
 - load image  from a server
 - upload an image
 - upload a lsst image and add all the mask

https://irsawebdev9.ipac.caltech.edu/dm-15157-image-refactor/firefly/

### CODE REVIEW

**Comment by lznakano** [comment]
It seems that wcs match is broken.  I compared with irsaviewer in irsa.ipac.caltech.edu.

**Comment by cwang2016** [comment]
The target seems missing for the following image search,
target: m31, wise all-sky single exposure, wise w1. 
please help check.

**Comment by robyww** [comment]
@cwang2016, @lznakano The WISE single projection were not being parsed correctly.  I think that is both the problem you saw.  I rebuilt it.

File: src/firefly/js/visualize/WebPlot.js [hunk 1]
```diff
@@ -311,6 +316,28 @@ function makePlotTemplate(plotId, plotType, asOverlay, imageCoordSys) {
 }
 
 
+function processAllAltWcs(header) {
+
+    const availableAry= alphabetAry.filter( (c) => header['CTYPE1'+c]);
+    if (isEmpty(availableAry)) return {};
+
+    return availableAry.reduce( (obj, altChar) => {
+        const processHeader= parseSpacialHeaderInfo(header, altChar);
```
Comment (cwang2016) [inline src/firefly/js/visualize/WebPlot.js]: is 'parseSpacialHeaderInfo' for both spacial WCS and alternative WCS axis description?

**Comment by lznakano** [comment]
I checked out the newest version and did many image related tests.  All worked as expected.  I like the new structure where all data related codes are placed in PlotData directory.  it is nicer, cleaner.  Put all all methods for wavelength calculations in Wavelength class is another nice clean up.  Very nice!!!

**Pull request merged**